packages/react/src/react-chat.ts

```diff
@@ -1,51 +1,25 @@
-import { StandardSchemaV1, Validator } from '@ai-sdk/provider-utils';
 import {
   AbstractChat,
+  AbstractChatInit,
   ChatState,
   ChatStatus,
-  ChatTransport,
   InferUIDataParts,
   UIDataPartSchemas,
   UIDataTypes,
   UIMessage,
-  UseChatOptions,
 } from 'ai';
 
 class ReactChatState<MESSAGE_METADATA, DATA_TYPES extends UIDataTypes>
   implements ChatState<MESSAGE_METADATA, DATA_TYPES>
 {
-  private messages: UIMessage<MESSAGE_METADATA, DATA_TYPES>[];
-  private status: ChatStatus = 'ready';
-  private error: Error | undefined = undefined;
+  messages: UIMessage<MESSAGE_METADATA, DATA_TYPES>[];
```

Comment:
for React this might need to be broken into a get/set pair so that the setter can spread the array -- TODO